lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -840,6 +840,17 @@ def ip_authentication?
         @options[:ip_authentication] == true
       end
 
+      def build_new_auth_purchase_order(action, money, payment_source, options)
+        build_new_order_xml(action, money, payment_source, options) do |xml|
+          add_payment_source(xml, payment_source, options)
+          add_address(xml, payment_source, options)
+          if @options[:customer_profiles]
+            add_customer_data(xml, payment_source, options)
+            add_managed_billing(xml, options)
+          end
+        end
+      end
+
       def build_new_order_xml(action, money, payment_source, parameters = {})
```

Comment:
It would be possible to further simplify this critical method, except that the Orbital API strictly enforces the _order_ of fields in the request. Therefore the various `add_something` method calls must remain in their seemingly random sequence.

Reply:
Usually XML follows a strict format that should adhere to some pre-determined XSD which most API's validate against. That said, I don't think this practice is applied to all API's that use XML.